docs/plugins/virtual_environment_docs/5-survey.md

```diff
@@ -0,0 +1,98 @@
+* However: If you have a scenario where you are actually creating a complicated widget directly in Qt, see `pytest-qt` for lots of tips, specifically `qtbot`.
+    - [pytest-qt](https://pytest-qt.readthedocs.io/en/latest/intro.html)
+    - [qtbot](https://pytest-qt.readthedocs.io/en/latest/reference.html?highlight=qtbot#module-pytestqt.qtbot)
+* Oftentimes, this comes down to knowing and/or learning the Qt API really well.  
+* Please see also the [In-depth guide to plugin testing](../../../../docs/plugins/testing_workshop_docs).
+## How to find the different signals or slots?
+Question: How can we find the different signals/slots we can connect callbacks to as the user interacts with the core napari interface e.g. creating/editing/deleting a `points` or `shapes` layer?
+
+Answer: 
+[https://napari.org/guides/stable/events_reference.html](https://napari.org/stable/guides/events_reference.html)  
+
+Granted, this is a work in progress. 
+
+For example, these events are emitted when the user interacts with the layer list: 
+```console  
+    Viewer.layers.events.inserted  
+    Viewer.layers.events.removed  
+    Viewer.layers.events.moved  
+    Viewer.layers.events.changed  
+    Viewer.layers.events.reordered  
+```    
+
+Getting an event when the user is editing the data inside a `points` or `shapes` layer (outside of the GUI interface) is complicated, because the user will be directly editing the native array object.
+
+## How do you avoid github tests failing?  
+Answer:  
+* First make sure all your tests are passing locally.  
+* After that, it's complicated. More background or context is needed to answer this question.  
+  
+## How do you make a process cancellable? 
+Question: How do you make a process cancellable to interrupt a method that is running in a for loop, for example?  
+
+Answer:  
+* In single-threaded python, use `Ctrl-C`  
+* In multithreaded python, there are many different patterns. Consider using a [generator-based thread worker].(https://napari.org/stable/guides/threading.html#generators-for-the-win)    
+
+## Are there testing environments in napari?
+Answer: Napari does not create or otherwise manage environments.  
+  
+## Introduction to npe2? Migrating to new plugin architecture?     
+Answer:  
+* The primary difference is in how plugins are discovered:  
+    - npe1 used decorators, requiring module import.  
+    - npe2 uses static manifests (`napari.yaml`), describing contributions without requiring import.  
+    - Your first plugin: https://napari.org/stable/plugins/first_plugin.html  
+  
+* Contributions Reference: https://napari.org/stable/plugins/contributions.html  
+* Guides for each type of contribution: https://napari.org/stable/plugins/guides.html  
+* Migration guide: https://napari.org/stable/plugins/npe2_migration_guide.html  
```

Comment:
```suggestion
Additional resources:
* [Contributions Reference](https://napari.org/stable/plugins/contributions.html)  
* [Guides for each type of contribution](https://napari.org/stable/plugins/guides.html)  
* [Migration guide](https://napari.org/stable/plugins/npe2_migration_guide.html)  
```

These bare links are clickable in GitHub as markdown, but not in the rendered html pages. This should make them work properly.